optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/MemberAccessorFactory.java

```diff
@@ -32,7 +34,12 @@
             SolverFactory.class.getSimpleName() + ".create...() method call.";
 
     public static MemberAccessor buildMemberAccessor(Member member, MemberAccessorType memberAccessorType,
-            Class<? extends Annotation> annotationClass) {
+            Class<? extends Annotation> annotationClass,
+            DomainAccessType domainAccessType) {
+        if (domainAccessType == DomainAccessType.GIZMO) {
+            return GizmoMemberAccessorImplementor.createAccessorFor(member, annotationClass);
```

Comment:
Very important:

1) Fail fast with a clear error message when they use `domainAccessType=GIZMO` without gizmo in the classpath.
A `Class.forName() based on a String after the if statement here probably does the trick. 
Suggestion for error message: "When using the domainAccessType (" + domainAccessType  + ") the classpath or modulepath must contain io.quarkus.gizmo:gizmo.\nMaybe add a dependency to io.quarkus.gizmo:gizmo.");

Here's what I get on spring-school-timetabling as-is with `optaplanner.solver.domainAccessType=GIZMO` in application.properties.

```
Caused by: java.lang.NoClassDefFoundError: io/quarkus/gizmo/ClassOutput
        at org.optaplanner.core.impl.domain.common.accessor.gizmo.GizmoMemberAccessorImplementor.createAccessorFor(GizmoMemberAccessorImplementor.java:163)
        at org.optaplanner.core.impl.domain.common.accessor.MemberAccessorFactory.buildMemberAccessor(MemberAccessorFactory.java:40)
        at org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor.processPlanningEntityPropertyAnnotation(SolutionDescriptor.java:472)
```

Reply:
2) Similar, if they do have gizmo in there, but have a non-public member, the exception message should be clear state which member isn't on a public member.

Here's what I get if I do that on the spring boot quickstart:

```
        Suppressed: java.lang.NullPointerException: null
                at org.optaplanner.core.impl.score.director.stream.ConstraintStreamScoreDirector.close(ConstraintStreamScoreDirector.java:105)
                at org.optaplanner.core.impl.score.DefaultScoreManager.updateScore(DefaultScoreManager.java:48)
                ... 51 common frames omitted
```

Hmm, wait a minute. In that stacktrace the real error got eaten by ConstraintStreamScoreDirector. @triceo .
So I am not sure the error message is already clean. Let me first fix that eating of the error message upstream.

Reply:
This is the exception that got eaten. Btw, it appears at runtime when going to localhost, not at bootstrap time when starting the server (nor at build time).

```
java.lang.IllegalAccessError: class com.example.schooltimetabling.domain.$optaplanner$__TimeTable$__lessonList tried to access private field com.example.schooltimetabling.domain.TimeTable.lessonList (com.example.schooltimetabling.domain.$optaplanner$__TimeTable$__lessonList is in unnamed module of loader 'OptaPlanner Gizmo MemberAccessor ClassLoader' @4a18e0f1; com.example.schooltimetabling.domain.TimeTable is in unnamed module of loader 'app')
	at com.example.schooltimetabling.domain.$optaplanner$__TimeTable$__lessonList.executeGetter($optaplanner$__TimeTable$__lessonList.zig:107)
	at org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor.extractMemberCollectionOrArray(SolutionDescriptor.java:1147)
```

This too should fail fast during bootstrap when the descriptors are created.

Reply:
The exception isn't eaten btw. The close exception is a "supressed exception", but that real exception is right above in the stacktrace. So no upstream issue.

Reply:
Done; thrown during bytecode generation (which is when the solverfactory is created)